### ISSUE TITLE
NPE in PartialParsingHelper

### ISSUE THREAD

**Issue by dhakehurst** [eclipse/xtext-core#219]
I'm experiencing a NPE in class PartialParsingHelper#87:
```
ReplaceRegion replaceRegion = tokenRegionProvider.getTokenReplaceRegion(insertChangeIntoReplaceRegion(oldRootNode, changedRegion), changedRegion);
```
The field tokenRegionProvider is injected optional, and it is null in my scenarion. There has to be a null guard.

**Comment by cdietrich** [comment]
@dhakehurst how did you manage it to provide null as TokenRegionProvider

**Comment by cdietrich** [comment]
fixed in Xtext 2.12
